### PR TITLE
Fix typo in ES service monitor name

### PR DESCRIPTION
**Pull request by hjiawei**
## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/monitor_test.go [hunk 1]
```diff
@@ -62,43 +57,24 @@ var _ = Describe("monitor rendering tests", func() {
...
+			{"prometheus-http-api", common.TigeraPrometheusNamespace, "", "v1", "Service"},
 		}
 
 		Expect(len(toCreate)).To(Equal(len(expectedResources)))
 
 		for i, expectedRes := range expectedResources {
 			obj := toCreate[i]
 			rtest.ExpectResource(obj, expectedRes.name, expectedRes.ns, expectedRes.group, expectedRes.version, expectedRes.kind)
-
-			if alertmanagerObj, ok := obj.(*monitoringv1.Alertmanager); ok {
```
Comment (hjiawei) [inline pkg/render/monitor_test.go:84]: These test cases are moved to its own `It` block in PR https://github.com/tigera/operator/pull/1374. Cleanup after merging from latest master.

**Comment by hjiawei** [comment]
Tested on GCP kubeadm and verified that the old ES ServiceMonitor will be deleted.

File: pkg/render/monitor.go [hunk 2]
```diff
@@ -308,6 +311,29 @@ func (mc *monitorComponent) serviceMonitorElasicsearch() *monitoringv1.ServiceMo
...
+// TODO Remove this object after we drop support for v3.8.
+func (mc *monitorComponent) serviceMonitorElasicsearchToDelete() *monitoringv1.ServiceMonitor {
+	return &monitoringv1.ServiceMonitor{
+		TypeMeta: metav1.TypeMeta{Kind: monitoringv1.ServiceMonitorsKind, APIVersion: MonitoringAPIVersion},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "elasticearch-metrics",
+			Namespace: common.TigeraPrometheusNamespace,
+			Labels:    map[string]string{"team": "network-operators"},
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
```
Comment (tmjd) [inline pkg/render/monitor.go:323]: I don't think any of the spec is needed here, deletes only need the name and namespace.

File: pkg/render/monitor.go [hunk 3]
```diff
@@ -308,6 +311,29 @@ func (mc *monitorComponent) serviceMonitorElasicsearch() *monitoringv1.ServiceMo
 	}
 }
 
+// TODO Remove this object after we drop support for v3.8.
```
Comment (tmjd) [inline pkg/render/monitor.go:314]: Could you include a note like "This is to delete a service that had been released with a typo in the name", just so it is obvious why we're deleting it and what was wrong with it?

**Review by tmjd — APPROVED** [review]
LGTM

**Pull request merged**